README.md

```diff
@@ -450,6 +454,65 @@ stdio to an in-memory [virtual terminal](https://github.com/hinshun/vt10x).
 
 For some examples, you can see any of the tests in this repo.
 
+### Using a mock
+
+Instead of calling the survey functions directly, you can create a survey struct and call the functions from there.
+
+```golang
+
+survey := survey.Survey{}
+
+response := false
+prompt := &survey.Confirm{
+    Message: "Do you like pie?",
+}
+survey.AskOne(prompt, &response)
```

Comment:
Now the big question is whether we should amend all other examples in the Readme to call `Ask` and `AskOne` on the Survey struct instead of on the package level? That would make everyone's code ultimately testable by default. Otherwise, the Readme will offer users usage examples of survey on the package level, but later the testing section will effectively tell them they need to rewrite their original code if they hope to mock Ask/AskOne calls. My hope for Survey is that it's testable by default without the need of changing the original implementation. @AlecAivazis Thoughts?

Reply:
I guess updating the readme should not be to much work, as it's just one or two lines of code changed in each example.

I left it the same for this pull request, because I wasn't sure how much of a "breaking" change you would accept for the docs.

Reply:
I updated the  examples in the readme, but I did not touch the text for now.

Some parts ("Running the Prompts", "Testing") likely still need updating.

Reply:
Hm, good question. I think keeping the examples as short as possible is probably the best. I'd like to avoid the `surveyor := survey.Surveyor{}` at the start of every example so either we can reference an undefined `surveyor` in the examples or we can just use the package-level functions. Either way, we definitely want to add a section about these two different approaches in the "Running the Prompts" section so it shouldn't be too confusing if we _do_ decide to update the tests to reference `surveyor.Ask` without explicitly showing its declaration.

Reply:
I think explicitly using `surveyor.Ask` in the examples is the way to go, because it makes the code testable by default (like @mislav noted). 

Referencing an undefined `surveyor` is probably not even a bad move, because it might move users directly to passing it to functions instead of creation a new instance everytime they want to ask a question.

I would recommend updating the first example to include a function where the surveyor is passed as an argument to show how it is done "correctly".